noir-projects/aztec-nr/aztec/src/macros/functions/mod.nr

```diff
@@ -238,22 +209,11 @@ comptime fn utility(f: FunctionDefinition) -> Quoted {
         );
     }
 
-    post_external_utility_checks(f);
-
-    // The abi export function is expected to be executed before the function is transformed.
-    let fn_abi_export = create_fn_abi_export(f);
-
-    transform_utility(f);
-
-    fn_abi_export
-}
-
-/// Utility functions cannot be used with the following modifiers: #[authorize_once], #[only_self], #[view],
-/// #[initializer], and #[noinitcheck]. Since we cannot enforce a specific ordering between these modifiers and
-/// #[external(...)], and we cannot access the #[external(...)] argument from within these modifiers' implementations
-/// (as accessing EXTERNAL_REGISTRY would require enforcing ordering), we perform these compatibility checks here in
-/// the utility macro.
-comptime fn post_external_utility_checks(f: FunctionDefinition) {
+    // Utility functions cannot be used with the following modifiers: #[authorize_once], #[only_self], #[view],
+    // #[initializer], and #[noinitcheck]. Since we cannot enforce a specific ordering between these modifiers and
+    // #[external(...)], and we cannot access the #[external(...)] argument from within these modifiers' implementations
```

Comment:
We could check this in the `#[aztec]` macro no? Go through all utility fns and check they're correct etc. Essentially first collect all atrtributes, put them in the registry, and then do all the work in one pass.